### ISSUE TITLE
Hosting: Add human readable error on API errors

### ISSUE THREAD

**Issue by gwwar** [Automattic/wp-calypso#37256]
<img width="1122" alt="Screen Shot 2019-11-01 at 3 05 38 PM" src="https://user-images.githubusercontent.com/1270189/68059300-3b90b780-fcb9-11e9-81ad-b085b5101395.png">

If we receive an API error after clicking on Enable SFTP or Open phpMyAdmin we should show a human readable error message to let folks know something went wrong. A dismissible error notice should do the trick.

https://wpcalypso.wordpress.com/devdocs/design/notice

cc @olaolusoga  @shaunandrews

**Comment by shaunandrews** [comment]
A standard notice _seems_ like it should do the trick, but its important to consider the length of the message and if there would be any associated action(s). A notice is best used for short, single sentence messages — and while the Notice component does support an action, having one alongside the dismiss button is undesirable.

Do you have to have specific examples of what API errors we expect to have, and how those would be worded?

**Comment by glendaviesnz** [comment]
@shaunandrews from what I can see we shouldn't get any specific expected errors at this stage. Any errors are likely to be unexpected due to network connection issues, etc. so something like the following may cover it:

<img width="973" alt="sftp-error" src="https://user-images.githubusercontent.com/3629020/68268436-0de1af00-00bb-11ea-977d-8a02579567af.png">

**Comment by shaunandrews** [comment]
I'm not sure we should use a Global Notice, as those tend to be more ephemeral — and really shouldn't be used for errors in my opinion. A standard Notice seems like it would do the trick.

As for the wording, I've asked @obi2020 to chime in, but my first reaction is that its too wordy. If we don't have a specific error to show, it also feels a little unhelpful.

Perhaps we accept those things and go with something like "Oops, something went wrong on our end. Please try again." — that is, we take the blame (you did nothing wrong!) and keep it simple.

Do we need them to refresh the page?

**Comment by obi2020** [comment]
I like the tone and direction of your edit @shaunandrews, and +1 to your question about refreshing, since we should state clearly what "try again" means.
